### DISCUSSION TITLE
is runs endpoint in the REST API misbehaving?

### DISCUSSION THREAD

**Discussion by gustavo-salazar** [community/community#14769] [Actions]
Hey there,

I'm trying to get information of the most recent run on a particular branch. So I'm using this URL: https://api.github.com/repos/ProteinsWebTeam/interpro7-client/actions/runs?branch=future

Right now the response I'm getting is giving me 7 runs and the most recent one is `"created_at": "2022-02-14T11:11:36Z"` however if I take the `branch` attribute put of the URL: https://api.github.com/repos/ProteinsWebTeam/interpro7-client/actions/runs
the most recent in that list with `"head_branch": "future"` is from `"created_at": "2022-04-11T16:21:55Z"`

Is the URL that I'm using to filter for that branch wrong somehow? or is this an issue in the REST APi, and if so, is this the right place to report it.

Thanks, 

Gustavo

**Comment by LixingSun** [comment]
Same issue here... 

For the request to get a workflow runs, if I do not provide any query parameter it could return the latest runs properly, but once adding `created` to do filtering, it seems only returned runs before April 12th in my case. 

I have tried several repos as well as regeneration of token. But unfortunately still didn't work for me yet...

**Comment by gustavo-salazar** [comment]
I think t'is working now. 
But quite annoying that after 10 days this never got an official response of at least acknowledgment, because now I think it's working but I'm not sure, which means I'm not trusting my CI results 😒

**Reply by LixingSun** [reply]
There was indeed some magic happened a few days ago. But from my testing although I could retrieve the latest runs now, but there were still quite some data missing in the past if try to pass parameter (in my case the period were around end of March).

So to me the gap is still there as long as pass parameter to request, and definitely it is still "unreliable"...

**Reply by gustavo-salazar** [reply]
Interesting. I didn't considered old builds because on my workflow I'm interested is in the most recent ones. 

BTW, there are a few reports similar to this one in github.community:

https://github.community/t/bug-we-are-having-problems-searching-workflow-runs-the-results-may-not-be-complete/244650
https://github.community/t/bug-github-api-list-workflow-runs-returns-0-runs-even-though-there-are-runs/165080/3

Still unsure where is the best place for this sort of issue to be reported.
